extensions/omni_httpd/master_worker.c

```diff
@@ -511,12 +511,19 @@ void master_worker(Datum db_oid) {
       strncpy(worker.bgw_library_name, MyBgworkerEntry->bgw_library_name, BGW_MAXLEN - 1);
       for (int i = 0; i < *num_http_workers; i++) {
         BackgroundWorkerHandle *handle;
-        RegisterDynamicBackgroundWorker(&worker, &handle);
-        pid_t worker_pid;
-        if (WaitForBackgroundWorkerStartup(handle, &worker_pid) == BGWH_POSTMASTER_DIED) {
-          goto terminate;
+        if (RegisterDynamicBackgroundWorker(&worker, &handle)) {
+          pid_t worker_pid;
+          if (WaitForBackgroundWorkerStartup(handle, &worker_pid) == BGWH_POSTMASTER_DIED) {
+            goto terminate;
+          }
+          cvec_bgwhandle_push(&http_workers, handle);
+        } else {
+          ereport(WARNING,
+              errcode(ERRCODE_INSUFFICIENT_RESOURCES),
+              errmsg("Could not start %s", worker.bgw_name),
+              errdetail("omni_httpd.http_workers = %d is higher than max_worker_processes = %d", *num_http_workers, max_worker_processes),
```

Comment:
Great! 👍 

Nitpick: since we know the numbers, why don't we check if http_workers is higher than allowed before issuing a WARNING.

Thinking a bit further, we can also look at addressing it early in `master_worker` – should we even try to allow num_http_workers to be higher than max_worker_process - 1 (at least one more is necessary  for the master worker itself). Note that this will not fully prevent the issue as the total number of workers in the cluster is out of our control and is subject to change. But it may prevent the problem cascading here for obvious clear-cut cases like 8/16.

Reply:
Yeah, I think that would be better.

Btw, shouldn't the defaults be conservative? For example Nginx defaults to 1 worker process https://nginx.org/en/docs/ngx_core_module.html#worker_processes and can be tuned to `auto` for consuming all available CPUs.

Wouldn't it make sense for omni_httpd to be `1` by default and use `0` as it's auto mode?

Reply:
Strange, I'm finding that `RegisterDynamicBackgroundWorker` always manages to start `max_worker_processes - 2`, not `max_worker_processes - 1`. 

It must be considering another worker besides the `omni_httpd` one?

Reply:
Ah, there's an additional `omni_startup` worker:
```
2024-07-06 12:51:10.665 -05 [607820] FATAL:  Can't register a dynamic background worker
2024-07-06 12:51:10.667 -05 [607814] LOG:  background worker "omni startup" (PID 607820) exited with exit code 
```

Reply:
For now I've harcoded the `2` and left it as TODO, assuming `omni_workers` might be able to handle that better.

This way at least we don't log WARNINGs unnecessarily.